File: Application/BackEnd/AmongUs-OurWay/Models/Friend.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AmongUs_OurWay.Models
{
    [Table("Friend")]
    public class Friend
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [DataType("int")]
        [Column("Id")]
        public int Id { get; set; }
        
        [DataType(DataType.Text)]
        [Column("User1Ref")]
        public string User1Ref { get; set; }
        
        [NotMapped]
        public virtual User User1 { get; set; }
        
        [DataType(DataType.Text)]
        [Column("User2Ref")]
        public string User2Ref { get; set; }
    }
}

File: Application/FrontEnd/src/app/models/profile/friend.ts
import { User } from "../user";
export class Friend {
    id: number;
    user1Ref: string;
    user1: User;
    user2Ref: string;

    constructor(){
        this.user1Ref = "";
        this.user1 = new User();
        this.user2Ref = "";
    }
}